flocker/node/test/test_loop.py

```diff
@@ -724,6 +726,43 @@ def test_status_update_while_sleeping_no_discovery(self):
             remaining_discover_calls - len(self.deployer.local_states),
             0)
 
+    def test_longer_sleep_when_converged(self):
+        """
+        When a convergence loop results in a ``NoOp`` the sleep is based on
+        that configured on the ``Deployer``.
+        """
```

Comment:
This test gets pretty deep into specific randomization implementation details of the loop which aren't really that important.  Some abstraction over those details would help the test be clearer and less fragile.  It might also help write a test that exercises the behavior we actually care about from the randomization (that the sleep interval varies to avoid load spikes due to herding) which don't appear to be tested currently.

:question: